test/e2e/framework/service/jig.go

```diff
@@ -507,7 +507,7 @@ func (j *TestJig) WaitForLoadBalancer(timeout time.Duration) (*v1.Service, error
 		}
 	}
 
-	return j.sanityCheckService(service, v1.ServiceTypeLoadBalancer)
+	return j.sanityCheckService(service, service.Spec.Type)
```

Comment:
I think it's not this line. Shouldn't this change be in "WaitForLoadBalancerDestroy" function?

[Well, probably it's correct here too, but in destroying it's also needed.]

Reply:
doh, yes, I meant to change it to `WaitForLoadBalancerDestroy`. (It's fine to hardcode it in `WaitForLoadBalancer`; there's no way we can reach the end of that function without erroring out unless the service is LoadBalancer.)